src/setup.rs

```diff
@@ -41,3 +45,24 @@ fn get_env_port() -> u16 {
 pub fn get_socket_addr() -> SocketAddrV4 {
     SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), get_env_port())
 }
+
+#[derive(Clone)]
+pub struct AppState {
+    connection_pool: Pool<Postgres>,
+}
+
+pub async fn create_app_state() -> AppState {
+    AppState {
+        pub connection_pool: database::get_connection_pool().await,
+    }
+}
```

Comment:
No clue why I suggested the pub field be done in the function instead of the struct definition, this is meaningless. Sorry!
```suggestion
#[derive(Clone)]
pub struct AppState {
    pub connection_pool: Pool<Postgres>,
}

pub async fn create_app_state() -> AppState {
    AppState {
        connection_pool: database::get_connection_pool().await,
    }
}
```